lightning/src/util/chacha20.rs

```diff
@@ -572,4 +598,35 @@ mod test {
 			assert_eq!(output, tv.keystream);
 		}
 	}
+
+	#[test]
+	fn get_single_block() {
+		// Test that `get_single_block` is equivalent to getting a block from a 12-byte nonce from the
+		// counter offset given by the remaining 4 bytes.
```

Comment:
The part of the sentence after "getting a block" is a bit difficult for me to follow because of the double "from". Should it be _using_ a 12-byte nonce? And what is meant by "given by the remaining 4 bytes" (i.e., what is "remaining" relative to)?

Reply:
"Remaining" is relative to the total 16 bytes of the nonce given to `get_single_block` (i.e. 12 bytes are used for the nonce, 4 bytes used for the counter). 

Reworded this, lmk if it's clearer